crates/rpc/rpc-types/src/eth/engine/payload.rs

```diff
@@ -397,14 +399,20 @@ impl PayloadStatusEnum {
 /// Various errors that can occur when validating a payload or forkchoice update.
 ///
 /// This is intended for the [PayloadStatusEnum::Invalid] variant.
-#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, thiserror::Error)]
 pub enum PayloadValidationError {
     /// Thrown when a forkchoice update's head links to a previously rejected payload.
     #[error("links to previously rejected block")]
     LinksToRejectedPayload,
     /// Thrown when a new payload contains a wrong block number.
     #[error("invalid block number")]
     InvalidBlockNumber,
+    /// Thrown when a new payload contains a wrong block hash.
+    #[error("invalid block hash")]
+    InvalidBlockHash { hash: H256 },
+    /// Thrown when a new payload violates consensus or execution rules
+    #[error("invalid block data")]
+    InvalidBlockData,
```

Comment:
I'd like to replace this variant with `Other(String)` variant, this helps with debugging.